controller/integration/testresource/testresource.go

```diff
@@ -4,47 +4,41 @@ package testresource
 
 import (
 	"context"
+	"sync"
 
 	"github.com/giantswarm/microerror"
-	"github.com/giantswarm/micrologger"
-	"k8s.io/client-go/kubernetes"
 )
 
 type Config struct {
-	K8sClient kubernetes.Interface
-	Logger    micrologger.Logger
-	Name      string
 }
 
 type Resource struct {
-	k8sClient kubernetes.Interface
-	logger    micrologger.Logger
-
 	createCount int
 	deleteCount int
-	name        string
+	mutex       sync.Mutex
```

Comment:
Adding the mutex here. This failed sometimes especially in the `parallel` test when two controllers got booted.